lib/src/migrators/module.dart

```diff
@@ -406,6 +407,12 @@ class _ModuleMigrationVisitor extends MigrationVisitor {
     for (var reference in references.sources.keys) {
       if (reference.span.sourceUrl != url) continue;
       var source = references.sources[reference];
+      if (source is ImportSource) {
+        if (source.ruleUrl == null) {
```

Comment:
Nit: combine with parent `if`

Reply:
Done. I seem to recall that Dart used to not recognize type guards that were part of complex if conditions, which is why I wrote it this way, but it looks like it actually does recognize it now.